dGame/LeaderboardManager.cpp

```diff
@@ -90,22 +92,36 @@ void LeaderboardManager::SaveScore(LWOOBJID playerID, uint32_t gameID, uint32_t
         const auto storedTime = result->getInt(1);
         const auto storedScore = result->getInt(2);
         auto highscore = true;
+        bool classicSurvivalScoring = Game::config->GetValue("classic_survival_scoring") == "1";
 
         switch (leaderboardType) {
             case ShootingGallery:
                 if (score <= storedScore)
                     highscore = false;
                 break;
             case Racing:
+                if (time >= storedTime)
+                    highscore = false;
+                break;
             case MonumentRace:
-                if (time > storedTime)
+                if (time >= storedTime)
                     highscore = false;
                 break;
             case FootRace:
                 if (time <= storedTime)
                     highscore = false;
                 break;
             case Survival:
+                if (classicSurvivalScoring) {
+                    if (time <= storedTime) { // Based on time (LU live)
+                        highscore = false;
+                    }
+                }
+                else {
+                    if (score <= storedScore) // Based on score (DLU)
+                        highscore = false;
+                }
+                break;
             case SurvivalNS:
                 if (score < storedScore || time >= storedTime)
```

Comment:
Pretty sure I found the logic to account for all scenarios. 

```suggestion
                if (!(score > storedScore || (time < storedTime && score >= storedScore)))
```